### PR TITLE
file: unzip introduced

### PR DESCRIPTION
**Pull request by tg44**
docs are missing, the basic test works, ideas welcome :)

### CODE REVIEW

**Comment by ennru** [comment]
Nice! Some docs would be required before merging.

**Comment by tg44** [comment]
I added docs, but can't test it, for some reason the google protobuff things are not compile on my machine.

File: file/src/main/scala/akka/stream/alpakka/file/impl/archive/ZipReaderSource.scala [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import akka.annotation.InternalApi
+import akka.stream.{Attributes, Outlet, SourceShape}
+import akka.stream.scaladsl.Source
+import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
+import akka.util.ByteString
+
+import java.io.{File, FileInputStream}
+import java.util.zip.{ZipEntry, ZipInputStream}
+
+case class ZipArchiveMetadata(name: String)
```
Comment (ennru) [inline file/src/main/scala/akka/stream/alpakka/file/impl/archive/ZipReaderSource.scala:17]: This class is user-facing API and should preferably live in `akka.stream.alpakka.file` instead. Add a `getName` for Java folks.

File: file/src/test/java/docs/javadsl/ArchiveTest.java [hunk 2]
```diff
@@ -106,6 +107,23 @@ public void flowShouldCreateZIPArchive() throws Exception {
     Map<String, ByteString> unzip = archiveHelper.unzip(resultFileContent);
 
     assertThat(inputFiles, is(unzip));
+    Path target = Files.createTempDirectory("alpakka-tar-");
```
Comment (ennru) [inline file/src/test/java/docs/javadsl/ArchiveTest.java:110]:
```suggestion
    Path target = Files.createTempDirectory("alpakka-zip-");
```

**Review by ennru — COMMENTED** [review]
Great. Some API tidiness suggestion.

**Comment by mkurz** [comment]
Thanks for working on this @tg44! Can you have a look at the suggestions @ennru made? I would love get this merged as well, thanks!

File: file/src/main/scala/akka/stream/alpakka/file/impl/archive/ZipReaderSource.scala (hunk 1 above)
Comment (tg44) [reply file/src/main/scala/akka/stream/alpakka/file/impl/archive/ZipReaderSource.scala]: Done!

File: file/src/test/java/docs/javadsl/ArchiveTest.java (hunk 2 above)
Comment (tg44) [reply file/src/test/java/docs/javadsl/ArchiveTest.java]: Done!

**Review by ennru — APPROVED** [review]
LGTM.

**Pull request merged**